tmt/result.py

```diff
@@ -356,119 +356,95 @@ def from_test_invocation(
+        else:
+            self.note = new_note
 
     def interpret_result(
             self,
-            interpret: ResultInterpret) -> 'Result':
+            interpret_test: ResultInterpret,
+            interpret_checks: dict[str, CheckResultInterpret]) -> 'Result':
         """
         Interpret result according to a given interpretation instruction.
 
         Inspect and possibly modify :py:attr:`result` and :py:attr:`note`
         attributes, following the ``interpret`` value.
 
-        :param interpret: how to interpret current result.
+        :param interpret_test: how to interpret current test result.
+        :param interpret_checks: how to interpret individual check results.
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
-        if interpret == ResultInterpret.CUSTOM:
-            return self
-
-        # Check for failed checks
-        checks_failed = any(check.result == ResultOutcome.FAIL for check in self.check)
-
-        if interpret == ResultInterpret.RESPECT:
-            if checks_failed:
-                self.result = ResultOutcome.FAIL
-                if self.note:
-                    self.note += ', check failed'
-                else:
-                    self.note = 'check failed'
+        if interpret_test == ResultInterpret.CUSTOM:
             return self
 
-        # Handle XFAIL for the overall test result
-        if interpret == ResultInterpret.XFAIL:
-            if self.result == ResultOutcome.PASS and checks_failed:
-                self.result = ResultOutcome.FAIL
-                if self.note:
-                    self.note += ', check failed'
-                else:
-                    self.note = 'check failed'
-            elif self.result == ResultOutcome.FAIL:
-                self.result = ResultOutcome.PASS
-                if self.note:
-                    self.note += ', expected failure'
-                else:
-                    self.note = 'expected failure'
-            elif self.result == ResultOutcome.PASS:
-                self.result = ResultOutcome.FAIL
-                if self.note:
-                    self.note += ', unexpected pass'
-                else:
-                    self.note = 'unexpected pass'
-            return self
-
-        # Handle individual check results
+        # First, handle individual check results
         for check in self.check:
-            check_result = CheckResultInterpret(check.result.value)
+            check_result = interpret_checks.get(check.name, CheckResultInterpret.RESPECT)
 
             if check_result == CheckResultInterpret.RESPECT:
                 if check.result == ResultOutcome.FAIL:
                     self.result = ResultOutcome.FAIL
-                    if self.note:
-                        self.note += ', check failed'
-                    else:
-                        self.note = 'check failed'
+                    self._set_note('check failed')
```

Comment:
This branch does look superfluous to me: `check.result` should be changed, not `self.result` - at this point, we're deciding the fate of a single check result, and the test result is beyond the horizon. The effect of this check result on the test result will be evaluated later. Which means, the outcome of the test did not change in any way, therefore no need for a note -> no need to check for `RESPECT` + `FAIL`.

Reply:
make sense